contracts/libraries/Ticks.sol

```diff
@@ -227,20 +230,15 @@ library Ticks {
                     pool.price,
                     liquidityPadded + uint256(pool.price) * uint256(cache.input)
                 );
-                /// @auditor - check tests to see if we need overflow handle
-                // if (!(nextTickPrice <= newPrice && newPrice < pool.price)) {
-                //     newPrice = uint160(PrecisionMath.divRoundingUp(liquidityPadded, liquidityPadded / pool.price + cache.input));
-                //  }
                 amountOut = DyDxMath.getDy(pool.liquidity, newPrice, uint256(pool.price), false);
                 cache.input = 0;
                 cache.cross = false;
                 pool.price = uint160(newPrice);
             } else { 
                 amountOut = DyDxMath.getDy(pool.liquidity, nextPrice, pool.price, false);
                 cache.input -= maxDx;
-                if (nextPrice == cache.crossPrice 
-                        && nextPrice != pool.price
-                        && cache.crossTick != TickMath.MIN_TICK) { cache.cross = true; }
+                if (nextPrice == cache.crossPrice
+                        && nextPrice != pool.price) { cache.cross = true; }
                 else cache.cross = false;
```

Comment:
Checking for a lack of change in `pool.price` tells us we've hit a tick boundary.